### PR TITLE
PR: fix stupidities

### PR DESCRIPTION
**Pull request by jernst**
Just a few things ...

### CODE REVIEW

File: src/feditest/nodedrivers/mastodon/__init__.py [hunk 1]
```diff
@@ -45,15 +46,15 @@ def _token_validate(candidate: str) -> str | None:
...
 class UserRecord:
     """
     Collects what we know of a user at a NodeWithMastodonAPI
     """
-    def __init__(self, userid: str, email: str, passwd: str):
-        self._userid = userid
-        self._email = email
-        self._passwd = passwd
-        self._mastodon_user_client = None
+    _userid : str
```
Comment (steve-bate) [inline src/feditest/nodedrivers/mastodon/__init__.py:54]: +1 on the dataclass change, but we should rename these fields without the leading underscore. It it's important, you can make the fields init-only using `@dataclass(frozen=True)` decorator option.

File: src/feditest/protocols/activitypub/__init__.py [hunk 2]
```diff
@@ -28,19 +28,23 @@ def __init__(self, uri: str):
...
         """
         if not self._json:
-            self._json = httpx.get(
+             # FIXME: this needs better error handling
+            r : httpx.Response = httpx.get(
                 self._uri,
                 follow_redirects=True,
                 verify=False,
                 headers={"Accept": "application/activity+json"})
+            if r.status_code != 200:
```
Comment (steve-bate) [inline src/feditest/protocols/activitypub/__init__.py:45]: FYI, `httpx.Response` has a `raise_for_status` method that will raise an exception on any HTTP error status code.

**Review by steve-bate — COMMENTED** [review]
Ideally, I'd like to see a set of fix that at least allow the Mastodon node unit tests to run. (I have local changes that modify those tests so they can be run against the WordPress node too. I can submit those after the Mastodon tests are fixed.) The unit tests don't ensure the actual feditests will pass, but if the unit tests don't pass it's almost guaranteed the feditests will also fail. On the other hand, while the tests are failing, I do appreciate the small incremental PRs. I just don't trust my old eyes to catch all the problems (if any).

**Comment by steve-bate** [comment]
I reassigned this PR back to you. I'm assuming you assigned it to me for review purposes? I'm already listed as a reviewer (and have submitted a review).

**Comment by jernst** [comment]
Good feedback. Changes made.

**Pull request merged**